lib/ddtrace/transport/trace_formatter.rb

```diff
@@ -165,6 +165,10 @@ def tag_sampling_priority!
 
       private
 
+      def partial?
```

Comment:
Can't comment on the line below but it turns out this comment I added because things were unclear to me and I had to retrace logic across three classes might just vanish by using this predicate.
 
 ```
        # when root span is not found, fall back to last span (partial flush)
        root_span || trace.spans.last
 ```